### PR TITLE
enrich LOI changes

### PR DESCRIPTION
**Pull request by jyotishk-eGov**

### CODE REVIEW

**Review by rushang7-eGov — CHANGES_REQUESTED** [review]
Look at the update Estimate enrichment and make changes accordingly.

File: backend/loi-service/src/main/java/org/egov/works/service/EnrichmentService.java [hunk 1]
```diff
@@ -46,6 +46,24 @@ public void enrichLOI(LetterOfIndentRequest request) {
         }
     }
 
+    public void enrichUpdateLOI(LetterOfIndentRequest request) {
+        RequestInfo requestInfo = request.getRequestInfo();
+        LetterOfIndent letterOfIndent = request.getLetterOfIndent();
+        String rootTenantId = letterOfIndent.getTenantId().split("\\.")[0];
+
+        letterOfIndent.setId(UUID.randomUUID());
```
[the remark this answers is not in the document]
Comment (jyotishk-eGov) [reply backend/loi-service/src/main/java/org/egov/works/service/EnrichmentService.java]: Removed this

File: backend/loi-service/src/main/java/org/egov/works/service/EnrichmentService.java [hunk 2]
```diff
@@ -46,6 +46,15 @@ public void enrichLOI(LetterOfIndentRequest request) {
         }
     }
 
+    public void enrichUpdateLOI(LetterOfIndentRequest request) {
+        RequestInfo requestInfo = request.getRequestInfo();
+        LetterOfIndent letterOfIndent = request.getLetterOfIndent();
+        String rootTenantId = letterOfIndent.getTenantId().split("\\.")[0];
```
Comment (rushang7-eGov) [inline backend/loi-service/src/main/java/org/egov/works/service/EnrichmentService.java:52]: Where are you using this variable? Remove unused variables.

File: backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java [hunk 3]
```diff
@@ -76,7 +76,7 @@ public List<LetterOfIndent> extractData(ResultSet rs) throws SQLException, DataA
                     .defectLiabilityPeriod(defectLiabilityPeriod)
                     .oicId(UUID.nameUUIDFromBytes(oicId.getBytes(StandardCharsets.UTF_8)))
                     .letterStatus(letterStatus)
-                    .additionalDetails(additionalDetails)
+                    .additionalDetails(additionalDetails.asText())
```
Comment (rushang7-eGov) [inline backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java:79]: Why asText()? Additional details is a json object, not a string.

**Review by rushang7-eGov — CHANGES_REQUESTED** [review]

File: backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java (hunk 3 above)
Comment (jyotishk-eGov) [reply backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java:79]: We are doing same way for search estimate.

**Review by rushang7-eGov — APPROVED** [review]

File: backend/loi-service/pom.xml [hunk 4]
```diff
@@ -93,8 +93,6 @@
         <dependency>
             <groupId>org.postgresql</groupId>
             <artifactId>postgresql</artifactId>
-            <scope>runtime</scope>
-            <version>${postgresql.version}</version>
```
Comment (rushang7-eGov) [inline backend/loi-service/pom.xml:97]: Why did you remove the version? Keep it.

**Review by rushang7-eGov — COMMENTED** [review]
Resolve the conflicts.

File: backend/loi-service/pom.xml (hunk 4 above)
Comment (jyotishk-eGov) [reply backend/loi-service/pom.xml:97]: The version we removed for the fix of additional details was not coming in proper format.
